sematic/types/types/datetime.py

```diff
@@ -0,0 +1,42 @@
+# Standard Library
+from dataclasses import dataclass
+from datetime import datetime
+
+
+@dataclass
```

Comment:
Sematic ❤️s dataclasses, though this would be better done with the native `datetime` class. I know we used a dataclass with `Link`, but that's just because there's no standard native python representation of them. For types that have native representations we prefer to stick with those. Here's how you could adapt to do that for this PR:
- don't define a new class/dataclass
- *do* keep this module
- in this module, add the following imports:
```python
from sematic.types.registry import (
    register_from_json_encodable,
    register_to_json_encodable,
    register_to_json_encodable_summary,
)
```
- use those as decorators on functions:
```python
@register_to_json_encodable_summary(datetime)
def _datetime_summary(value: datetime, _) -> str:
    # ...

@register_to_json_encodable(datetime)
def _datetime_to_encodable(value: datetime, type_: typing.Type[datetime]) -> str:
    # ...


@register_from_json_encodable(datetime)
def _datetime_from_encodable(value: str, type_: typing.Type[datetime]) -> datetime:
    # ...
```
`_datetime_to_encodable` and `_datetime_summary` can probably just return the ISO string representation for the datetime, while `_datetime_from_encodable` can deserialize from the ISO string representation and return a `datetime`

Reply:
See b44b5052cd6cc9cbd07ce4c233259a70d17c12de. I have implemented the methods you suggested. Let me know if this is what you had in mind 🙂.

Also, I have updated the tests according to the new DateTime representation. 🙂.

<img width="873" alt="Screenshot 2022-11-05 at 12 43 58" src="https://user-images.githubusercontent.com/32076170/200119100-8ffa561e-0a89-4045-98e1-e95318774d78.png">